### DISCUSSION TITLE
use userland fuse

### DISCUSSION THREAD

**Discussion by asynxc** [juicedata/juicefs#2758] [Ideas]
Hello, userland fuse is a convenient way for security reasons and mostly portability that allow users to setup JuiceFs in non privileged contexts (containers for example)

there's a good go library that have a good implementation of fuse matching the kernel api in userland.
i don't know if there's any other library out there

https://github.com/bazil/fuse

Is this a possible thing ? Thnx

**Comment by xiaogaozi** [comment]
JuiceFS use https://github.com/hanwen/go-fuse

**Reply by asynxc** [reply]
Hello, thanks for replying @xiaogaozi 

I Apologies for not being clear, actually i saw that JuiceFs uses go-fuse under the hood, and i also saw that in order use JuiceFs in a container, it must be run as privileged. which is a reasonably constraint since go-fuse binds to fuse kernel module.

The snippet that confirm my assumption, from here https://juicefs.com/docs/community/juicefs_on_docker
`sudo docker run -d --name nginx \
  -v /mnt/jfs/html:/usr/share/nginx/html \
  -p 8080:80 \
  --privileged=true \
  nginx-with-jfs`

So my question actually is, is it possible to use JuiceFs in a container without `--privileged=true` ?
if not, a userland fuse would't be a solution ?

Thanks ^^
